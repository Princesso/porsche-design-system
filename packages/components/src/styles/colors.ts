import type { Theme } from '@porsche-design-system/utilities-v2';
import { isThemeDark } from '../utils';

type ThemedColor =
  | 'primaryColor'
  | 'primaryColorDarken'
  | 'backgroundColor'
  | 'backgroundSurfaceColor'
  | 'contrastLowColor'
  | 'contrastMediumColor'
  | 'contrastHighColor'
  | 'contrastHighColorDarken'
  | 'hoverColor'
  | 'hoverColorDarken'
  | 'activeColor'
  | 'focusColor'
  | 'disabledColor'
  | 'errorColor'
  | 'errorColorDarken'
  | 'errorSoftColor'
  | 'successColor'
  | 'successColorDarken'
  | 'successSoftColor'
  | 'warningColor'
  | 'warningSoftColor'
  | 'infoColor'
  | 'infoSoftColor';

export type ThemedColors = { [key in ThemedColor]: string };

/* Auto Generated Start */
const themeLight = {
  primaryColor: '#010205',
  primaryColorDarken: '#000000',
  backgroundColor: '#FFF',
  backgroundSurfaceColor: '#EEEFF2',
  contrastLowColor: '#D8D8DB',
  contrastMediumColor: '#949598',
  contrastHighColor: '#535457',
  contrastHighColorDarken: '#353638',
  hoverColor: '#D8D8DB',
  hoverColorDarken: '#b8b8be',
  activeColor: '#D8D8DB',
  focusColor: '#0A0AFF',
  disabledColor: '#D8D8DB',
  errorColor: '#E7323B',
  errorColorDarken: '#c51720',
  errorSoftColor: '#FCE8E9',
  successColor: '#32B85B',
  successColorDarken: '#258843',
  successSoftColor: '#EBFAF0',
  warningColor: '#FECC1B',
  warningSoftColor: '#FFF9E6',
  infoColor: '#1E5BEB',
  infoSoftColor: '#E8EEFD'
};

const themeDark = {
  primaryColor: '#FBFCFF',
  primaryColorDarken: '#beceff',
  backgroundColor: '#0E0E12',
  backgroundSurfaceColor: '#212225',
  contrastLowColor: '#404044',
  contrastMediumColor: '#7E7F82',
  contrastHighColor: '#AFB0B3',
  contrastHighColorDarken: '#909195',
  hoverColor: '#404044',
  hoverColorDarken: '#222224',
  activeColor: '#404044',
  focusColor: '#0A0AFF',
  disabledColor: '#404044',
  errorColor: '#CB3333',
  errorColorDarken: '#9a2727',
  errorSoftColor: '#290A0A',
  successColor: '#00C77E',
  successColorDarken: '#008a57',
  successSoftColor: '#003320',
  warningColor: '#DDB84B',
  warningSoftColor: '#2B2208',
  infoColor: '#027FFC',
  infoSoftColor: '#001A33'
};

const themes = {
  'light': themeLight,
  'dark': themeDark
};
/* Auto Generated End */

export const getThemedColors = (theme: Theme): ThemedColors => {
  return themes[theme];
};

export const getInvertedThemedColors = (theme: Theme): ThemedColors => {
  return getThemedColors(isThemeDark(theme) ? 'light' : 'dark');
};
