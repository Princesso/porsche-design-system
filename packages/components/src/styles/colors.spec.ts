import { getThemedColors } from './colors';
import * as a11yUtils from '../utils/a11y/a11y';

describe('getThemedColors()', () => {
  it.each<Parameters<typeof getThemedColors>>([['light'], ['dark']])(
    'should return correct colors for theme: %s',
    (theme) => {
      expect(getThemedColors(theme)).toMatchSnapshot();
    }
  );
  it.each<Parameters<typeof getThemedColors>>([['light'], ['dark']])(
    'should return correct partial high contrast mode scheme colors for theme: %s',
    (theme) => {
      jest.spyOn(a11yUtils, 'highContrastMode').mockReturnValue(true);
      expect(getThemedColors(theme)).toMatchSnapshot();
    }
  );
});
