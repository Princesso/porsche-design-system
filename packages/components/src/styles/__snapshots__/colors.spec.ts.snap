// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getThemedColors() should return correct colors for theme: dark 1`] = `
{
  "activeColor": "#7E7F82",
  "backgroundColor": "#0E0E12",
  "backgroundSurfaceColor": "#212225",
  "contrastHighColor": "#AFB0B3",
  "contrastHighColorDarken": "#909195",
  "contrastLowColor": "#404044",
  "contrastMediumColor": "#7E7F82",
  "disabledColor": "#7E7F82",
  "errorColor": "#CB3333",
  "errorColorDarken": "#9a2727",
  "errorSoftColor": "#290A0A",
  "focusColor": "#0A0AFF",
  "hoverColor": "rgba(144, 144, 153, 0.25)",
  "hoverColorDarken": "#71717b",
  "infoColor": "#027FFC",
  "infoSoftColor": "#001A33",
  "primaryColor": "#FBFCFF",
  "primaryColorDarken": "#beceff",
  "successColor": "#00C77E",
  "successColorDarken": "#008a57",
  "successSoftColor": "#003320",
  "warningColor": "#DDB84B",
  "warningSoftColor": "#2B2208",
}
`;

exports[`getThemedColors() should return correct colors for theme: light 1`] = `
{
  "activeColor": "#949598",
  "backgroundColor": "#FFF",
  "backgroundSurfaceColor": "#EEEFF2",
  "contrastHighColor": "#535457",
  "contrastHighColorDarken": "#353638",
  "contrastLowColor": "#D8D8DB",
  "contrastMediumColor": "#949598",
  "disabledColor": "#949598",
  "errorColor": "#E7323B",
  "errorColorDarken": "#c51720",
  "errorSoftColor": "#FCE8E9",
  "focusColor": "#0A0AFF",
  "hoverColor": "rgba(83, 83, 83, 0.25)",
  "hoverColorDarken": "#343434",
  "infoColor": "#1E5BEB",
  "infoSoftColor": "#E8EEFD",
  "primaryColor": "#010205",
  "primaryColorDarken": "#000000",
  "successColor": "#32B85B",
  "successColorDarken": "#258843",
  "successSoftColor": "#EBFAF0",
  "warningColor": "#FECC1B",
  "warningSoftColor": "#FFF9E6",
}
`;
